app/models/manageiq/providers/autosde/inventory/parser/storage_manager.rb

```diff
@@ -181,11 +181,30 @@ def storage_service_resource_attachments
     end
   end
 
-  # This method changes capability name field from 'abstract_capability' to 'name'
-  def parse_capabilities(capabilities, field_name)
-    caps = JSON.parse(capabilities)
-    caps.each do |capability|
-      capability['name'] = capability.delete(field_name)
+  def parse_ems_capabilities(capabilities)
+    return capabilities unless capabilities.is_a?(Array)
+
+    capabilities.each_with_object(Hash.new { |k, v| k[v] = []}) do |capability, result|
+      result[capability.abstract_capability] << {"uuid" => capability.uuid, "value" => capability.value}
+    end
+  end
+
+  def parse_possible_capabilities(capabilities, field_name)
+    caps_hash = {}
+    JSON.parse(capabilities).each do |capability|
+      name = capability[field_name]
+      (caps_hash[name] ||= []) << capability["value"]
     end
+    caps_hash
```

Comment:
Again style wise we try not to do
```
result = {}
loop do
  result << something
end
result
```

Better to use `.to_h` or `.each_with_object` depending on your use case.  Can you tackle this in a follow-up immediately after this?